File: situational-awareness/frontend/src/types/report.ts
export type Report = {
  id: string;
  scope: "job" | "asset";
  scope_id: string;
  summary_md: string;
  risk_overview_json: Record<string, number>;
  analysis_json: Record<string, unknown>;
  created_at: string;
};

export type GenerateReportResponse = {
  task_id: string;
  status: string;
};

File: situational-awareness/backend/app/schemas/report.py
from datetime import datetime

from pydantic import BaseModel

from app.db.models.enums import ReportScope
from app.schemas.common import ORMModel


class GenerateReportResponse(BaseModel):
    task_id: str
    status: str


class ReportRead(ORMModel):
    id: str
    scope: ReportScope
    scope_id: str
    summary_md: str
    risk_overview_json: dict
    analysis_json: dict
    created_at: datetime
